### ISSUE TITLE
[BUG] pcidevices controller fails while setting up sriov network devices

### ISSUE THREAD

**Issue by ibrokethecloud** [harvester/harvester#6502]
**Describe the bug**

On certain machines where nic names may have an upper case character the setup of sriov network devices fails since camelcase is not allowed in kubernetes object names.

Users will notice the following error in the pcidevices controller logs

```
harvester-pcidevices-controller-cmwmb agent time="2024-09-04T10:58:52Z" level=error msg="error syncing 'amp2': handler reconcile-pcidevices: error setting up sriov devices for node amp2: error creating device amp2-enP3p3s0f0 on node amp2: SRIOVNetworkDevice.devices.harvesterhci.io \"amp2-enP3p3s0f0\" is invalid: metadata.name: Invalid value: \"amp2-enP3p3s0f0\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), requeuing"
```
**To Reproduce**

**Expected behavior**

sriov network device object should be creaated

**Support bundle**

**Environment**
 - Harvester ISO version: v1.3.1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Arm Ampere Ultra board with Mellanox nics

**Additional context**

**Comment by TachunLin** [comment]
This issue require specific ARM based machine to be connected to a working switch.

**Comment by TachunLin** [comment]
Thanks @ibrokethecloud for preparing the special bare-metal machine with NICs named with camel case


Verified **fixed** on `v1.4.0-rc5`. Close this issue.


### Result
<details><summary>$\color{green}{\textsf{PASS}}$ Network interfaces have camel case name can be created as the SR-IOV network devices  $~~$</summary>

  1. On the ARM based machine, when we have network interface named in camel case 

  ```
  rancher@harvester-pxe-1:~> ip a
  1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
      link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
      inet 127.0.0.1/8 scope host lo
         valid_lft forever preferred_lft forever
  2: eno1: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN group default qlen 1000
      link/ether 18:c0:4d:0d:9c:e6 brd ff:ff:ff:ff:ff:ff
      altname enP3p3s0f0
  3: eno2: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN group default qlen 1000
      link/ether 18:c0:4d:0d:9c:e7 brd ff:ff:ff:ff:ff:ff
      altname enP3p3s0f1
  4: enP1p1s0f0np0: <BROADCAST,MULTICAST,SLAVE,UP,LOWER_UP> mtu 1500 qdisc mq master mgmt-bo state UP group default qlen 1000
      link/ether 0c:42:a1:49:d0:e4 brd ff:ff:ff:ff:ff:ff
  5: enP1p1s0f1np1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
      link/ether 0c:42:a1:49:d0:e5 brd ff:ff:ff:ff:ff:ff
  ```
  2. The camel case NIC can be created as SR-IOV device without prompting failure in `pcidevices-controller` log


  ![image](https://github.com/user-attachments/assets/7283ca77-b1d5-4c80-a783-649ab53d937a)


 </details>



### Test Information
* Test Environment: Single node harvester ARM based bare-metal machine on Equinix platform
* Harvester version: `v1.4.0-rc5`

### Verify Steps
<details><summary> Create the SR-IOV network devices on camel case NICs </summary>

  1. Prepare a bare-metal machine 
  2. Which have network interface support SR-IOV (FV)
  3. The NICs have named with camel case (e.g `enp1p1s0f1np1`)
  4. Enable the `pcidevices-controller` in addons page
  5. Open the Advanced -> SR-IOV Network Devices page
  6. Check can create the SR-IOV network device well without error

</details>
